IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate.Tests/ConditionalWaitTests.cs

```diff
@@ -39,9 +184,80 @@ private bool TestReturnDelegate()
             throw new Exception();
         }
 
-        private void TestNullRefException()
+        private void ThrowNullRefException()
         {
             throw new NullReferenceException();
         }
+
+        private void Equals(int expected, int value)
+        {
+            if (value == expected)
+            {
+                return;
+            }
+            throw new EqualException(expected, value);
+        }
+
+        private void IterateResults(int secondsToFail = 1, int numberOfDifferentExceptions = 1)
+        {
+            if (_End == DateTime.MinValue)
+            {
+                _End = DateTime.Now.AddSeconds(secondsToFail);
+            }
+
+            while (DateTime.Now < _End)
```

Comment:
Basically I just needed some sort of method to throw multiple different exceptions. Basically, I want to replicate ui testing behavior, where a framework (be it selenium, WinAppDriver, or etc.) can throw an element not found error, then an element is not interactable error, then etc. until the element hits the expected state. At some point it may pass successfully, but until that point, there may be one of many expected exceptions thrown. Thoughts on if there's a better way to do this?

Reply:
Although I did also notice some errors in that method that was causing it to not throw all of the exceptions I was expecting... WHICH because of the issue you pointed out with the Assert.IsNotNull(OfType) things, I never caught. :( I think I at least have it fixed to where things are working as I was intending now.